velox/functions/prestosql/aggregates/tests/MaxSizeForStatsTest.cpp

```diff
@@ -0,0 +1,261 @@
+       "\"$internal$max_data_size_for_stats\"(c2)",
+       "\"$internal$max_data_size_for_stats\"(c3)",
+       "\"$internal$max_data_size_for_stats\"(c4)",
+       "\"$internal$max_data_size_for_stats\"(c5)",
+       "\"$internal$max_data_size_for_stats\"(c6)",
+       "\"$internal$max_data_size_for_stats\"(c7)",
+       "\"$internal$max_data_size_for_stats\"(c8)",
+       "\"$internal$max_data_size_for_stats\"(c9)"},
+      "SELECT * FROM (VALUES (1,2,4,8,4,8,1,4,16))");
+}
+
+// Input:
+// c0(array(bigint))
+// [1,2,3,4,5]
+// []
+// [1,2,3]
+// Query:
+// select aggr(c0)
+// Output
+// 44
+TEST_F(MaxSizeForStatsTest, arrayGlobalAggregate) {
+  auto vectors = {makeRowVector({makeArrayVector<int64_t>({
+      {1, 2, 3, 4, 5},
+      {},
+      {1, 2, 3},
+  })})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 44");
+}
+
+// Input:
+// c0(map(tinyint,integer))
+// map(array(1,1,1,1,1),array(1,1,1,1,1))
+// map()
+// map(array(1,1,1), array(1,1,1))
+// Query:
+// select aggr(c0)
+// Output
+// 29
+TEST_F(MaxSizeForStatsTest, mapGlobalAggregate) {
+  auto vectors = {makeRowVector({makeMapVector<int8_t, int32_t>(
+      {{{1, 1}, {1, 1}, {1, 1}, {1, 1}, {1, 1}},
+       {},
+       {{1, 1}, {1, 1}, {1, 1}}})})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 29");
+}
+
+// Input:
+// c0( row(array(bigint), map(tinyint,integer))
+// row(aray[1,2,3,4,5], map(array(1,1,1,1,1),array(1,1,1,1,1)))
+// row(array[], map())
+// row(array[1,2,3], map(array(1,1,1), array(1,1,1)))
+// Query:
+// select aggr(c0)
+// Output
+// 77
+TEST_F(MaxSizeForStatsTest, rowGlobalAggregate) {
+  auto vectors = {makeRowVector({makeRowVector(
+      {makeArrayVector<int64_t>({
+           {1, 2, 3, 4, 5},
+           {},
+           {1, 2, 3},
+       }),
+       makeMapVector<int8_t, int32_t>(
+           {{{1, 1}, {1, 1}, {1, 1}, {1, 1}, {1, 1}},
+            {},
+            {{1, 1}, {1, 1}, {1, 1}}})})})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 77");
+}
+
+// Input:
+// c0(varbinary)
+// "buf"
+// ""
+// "hello"
+// query:
+// select aggr(c0)
+// output:
+// 9
+TEST_F(MaxSizeForStatsTest, varbinaryGlobalAggregate) {
+  VectorPtr varbinary_vector = BaseVector::create(VARBINARY(), 3, pool());
+  auto flat_vector = varbinary_vector->asFlatVector<StringView>();
+  StringView sv1{"buf"};
+  StringView sv2{""};
+  StringView sv3{"henlo"};
+  flat_vector->set(0, sv1);
+  flat_vector->set(1, sv2);
+  flat_vector->set(2, sv3);
+
+  auto vectors = {makeRowVector({varbinary_vector})};
+  testAggregations(
+      vectors, {}, {"\"$internal$max_data_size_for_stats\"(c0)"}, "SELECT 9");
+}
+
+// Input:
+// c0(varchar)
+// "{1, 2, 3, 4, 5}"
+// "{}"
+// "{1, 2, 3}"
+// query:
+// select aggr(c0)
+// output:
+// 19
+TEST_F(MaxSizeForStatsTest, varcharGlobalAggregate) {
+  auto vectors = {makeRowVector({makeFlatVector<StringView>({
```

Comment:
I wonder how does this work. I'd expect string values here, not integers.

Reply:
So we have 3 rows of strings values:
"{1, 2, 3, 4, 5}",
      "{}",
      "{1, 2, 3}",
The first row (with 15 characters) is the longest and its serialized size is 15 + 4 (overhead storage for variable length data types such varbinary and varchar)

Reply:
Thank you for explaining. I missed the quotes around {1, 2, 3, 4, 5}. Would it be easier to read this code if strings didn't look like arrays of integers, e.g. "apple",  "tomato".